functions/Test-DbaRepLatency.ps1

```diff
@@ -0,0 +1,230 @@
+
+                # Call the PostTracerToken method. This method inserts a tracer token into the publication's Transaction log.
+                $transPub.PostTracerToken() | Out-Null
+            }
+
+            ##################################################################################
+            ### Determine Latency and validate connections for a transactional publication ###
+`           ##################################################################################
+
+            $repServer = New-Object Microsoft.SqlServer.Replication.ReplicationServer
+
+            # Set the Name and DatabaseName properties for the Replication Server, and set the ConnectionContext property to the connection created in step 1.
+            $repServer.ConnectionContext = $server.ConnectionContext.SqlConnectionObject
+
+            $distributionServer = $repServer.DistributionServer
+            $distributionDatabase = $repServer.DistributionDatabase
+
+            # Step 1: Connect to the distributor
+            Write-Message -Level Verbose -Message "Connecting to Distributor"
+
+            try {
+                $distServer = Connect-SqlInstance -SqlInstance $DistributionServer -SqlCredential $SqlCredential -MinimumVersion 9
+            }
+            catch {
+                Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $DistributionServer -Continue
+            }
+
+            foreach($publication in $publicationNames) {
+
+                $pubMon = New-Object Microsoft.SqlServer.Replication.PublicationMonitor
+
+                $pubMon.Name = $publication.PublicationName
+                $pubMon.DistributionDBName = $distributionDatabase 
+                $pubMon.PublisherName = $publication.Server
+                $pubMon.PublicationDBName = $publication.Database
+
+                $pubMon.ConnectionContext = $distServer.ConnectionContext.SqlConnectionObject;
+
+
+                # Call the LoadProperties method to get the properties of the object. If this method returns false, either the publication monitor properties in Step 3 were defined incorrectly or the publication does not exist.
+                if(!$pubMon.LoadProperties()) {
+                    Stop-Function -Message "LoadProperties() failed. The publication does not exist." -Continue
+                }
+
+                $tokenList = $pubMon.EnumTracerTokens()
+
+                if(!$DisplayTokenHistory) {
+                    $tokenList = $tokenList[0]
+                }
+
+
+                foreach($token in $tokenList) {
+
+                    $tracerTokenId = $token.TracerTokenId
+
+                    $tokenInfo = $pubMon.EnumTracerTokenHistory($tracerTokenId)
+
+                    $timer = 0;
+                
+                    while($tokenInfo.Tables[0].Rows[0].subscriber_latency -eq [System.DBNull]::Value) {
+                        if($TimeToLive -and ($timer -gt $TimeToLive)) {
+                            break;
```

Comment:
breaks are not allowed because they are too breaky. Please use Stop-Function with -Continue instead.